pkg/apis/monitoring/v1/types.go

```diff
@@ -150,6 +150,52 @@ func (pc *ProxyConfig) Validate() error {
 	return nil
 }
 
+type HTTPHeader struct {
+	// Name of the HTTP header.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// References to secret keys holding the header values.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:Required
+	SecretRefs []v1.SecretKeySelector `json:"secretRefs"`
```

Comment:
from https://github.com/prometheus-operator/prometheus-operator/issues/7264#issuecomment-2582564158, I'd recommend that we define/use our own SecretKeySelector struct to have proper validation.

Reply:
Thanks @simonpasquier 
Can the custom SecretKeySelector structure be implemented uniformly in the next PR to replace it?
I would be happy to complete this PR. WDYT?